algorithms.py

```diff
@@ -584,13 +585,37 @@ def hudson_recombination_event(self):
             y.next = None
             y.right = k
             self.L.increment(y.index, k - z.right)
+            if self.full_arg:
+                self.store_node(y.population)
+                u = len(self.tables.nodes) - 1
+                self.store_edge(y.left, y.right, u, y.node)
+                y.node = u
+                while y.prev is not None:
+                    y = y.prev
+                    y.node = u
         else:
             # split the link between x and y.
             x.next = None
             y.prev = None
             z = y
+            if self.full_arg:
+                self.store_node(x.population)
+                u = len(self.tables.nodes) - 1
```

Comment:
This is identical to the section above. To avoid repetition, maybe we could add a method ``store_edges_left`` function which does this? (And corresponding right for below?)

Reply:
That would definitely be possible. Each of the store_edge loops gets called a couple of times at present.